### PR TITLE
Hekate

### PR DESCRIPTION
**Pull request by kolektiv**
Now uses Hekate graph lib for the basis of Machine, both for the runtime stuff and dependency analysis. This has resulted in quite a lot less code in machine, and a generally nicer execution model. Pull request initially for visibility (though it's ready to merge, it needs some upstream changes merged in...)

### CODE REVIEW

**Comment by panesofglass** [comment]
YES! We need to look into adding NuGet package generation for multiple target frameworks, btw, for all of Aether, Hekate, and Freya.

**Comment by kolektiv** [comment]
Yup, agreed on that. Let's chat about that later :)

**Comment by kolektiv** [comment]
Can we merge this? I think it's reasonable...

File: src/Freya.Core/Functions.fs [hunk 1]
```diff
@@ -41,24 +41,24 @@ let modM f =
 
 /// Gets part of the Core State within a Core monad using an Aether lens
 let getLM l = 
-    getL l <!> getM
+    Lens.get l <!> getM
```
Comment (panesofglass) [inline src/Freya.Core/Functions.fs]: Thank you for the full names!

**Comment by kolektiv** [comment]
Awesome :) on that full names front, i've been moving in that direction with other libraries. The getLM, setPLM etc. methods in Freya could well be considered candidates for something similar... Freya.getLens, Freya.setLensPartial, etc. would be the change I imagine - I've done something similar in Chiron, and the verbosity isn't too jarring - and a lot more discoverable...

**Comment by panesofglass** [comment]
Yes, that would be nice. :)

**Pull request merged**